src/org/opendatakit/briefcase/operations/PullFormFromCentral.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.operations;
+
+import static java.util.stream.Collectors.toList;
+import static org.opendatakit.briefcase.operations.Common.CREDENTIALS_EMAIL;
+import static org.opendatakit.briefcase.operations.Common.CREDENTIALS_PASSWORD;
+import static org.opendatakit.briefcase.operations.Common.FORM_ID;
+import static org.opendatakit.briefcase.operations.Common.MAX_HTTP_CONNECTIONS;
+import static org.opendatakit.briefcase.operations.Common.PROJECT_ID;
+import static org.opendatakit.briefcase.operations.Common.SERVER_URL;
+import static org.opendatakit.briefcase.operations.Common.STORAGE_DIR;
+import static org.opendatakit.briefcase.reused.http.Http.DEFAULT_HTTP_CONNECTIONS;
+
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import org.opendatakit.briefcase.model.BriefcasePreferences;
+import org.opendatakit.briefcase.model.FormStatus;
+import org.opendatakit.briefcase.model.FormStatusEvent;
+import org.opendatakit.briefcase.model.RemoteFormDefinition;
+import org.opendatakit.briefcase.pull.central.PullFromCentral;
+import org.opendatakit.briefcase.reused.BriefcaseException;
+import org.opendatakit.briefcase.reused.Optionals;
+import org.opendatakit.briefcase.reused.http.CommonsHttp;
+import org.opendatakit.briefcase.reused.http.Credentials;
+import org.opendatakit.briefcase.reused.http.Http;
+import org.opendatakit.briefcase.reused.http.response.Response;
+import org.opendatakit.briefcase.reused.job.JobsRunner;
+import org.opendatakit.briefcase.reused.transfer.CentralServer;
+import org.opendatakit.briefcase.transfer.TransferForms;
+import org.opendatakit.briefcase.util.FormCache;
+import org.opendatakit.common.cli.Args;
+import org.opendatakit.common.cli.Operation;
+import org.opendatakit.common.cli.Param;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PullFormFromCentral {
```

Comment:
Is it still conventional to use nouns for classes? If so, how about CentralFormPuller?

Reply:
`CentralFormPuller` wouldn't be correct because this class statically describes a pull operation but it doesn't actually perform it (that's the `JobsRunner`'s work).

I consider this class to be part of a group of domain actions. The complete name would be `PullFormFromCentralAction` but it's longer and feels redundant.